File: src/main/java/com/example/socialnetworkingapp/model/job/JobRequest.java
package com.example.socialnetworkingapp.model.job;

import com.example.socialnetworkingapp.model.tags.Tag;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
public class JobRequest {

    private String title;
    private String location;
    private JobType jobType;
    private ExperienceLevel experienceLevel;
    private String info;
    private List<Tag> tags;
}

File: socialnetworkingapp-front/src/app/job/job-request.ts
import {Tag} from "../tags/Tag";
import {JobType} from "./jobType";
import {ExperienceLevel} from "./experienceLevel";

export class JobRequest {

  title: string;
  location: string;
  jobType: JobType;
  experienceLevel: ExperienceLevel;
  info: string;
  tags: Tag[];

  constructor(title: string, location: string, jobType: JobType, experienceLevel: ExperienceLevel, info: string, tags: Tag[]) {
    this.title = title;
    this.location = location;
    this.jobType = jobType;
    this.experienceLevel = experienceLevel;
    this.info = info;
    this.tags = tags;
  }
}
